giskard/core/core.py

```diff
@@ -101,7 +101,7 @@ class ModelMeta:
 @dataclass
 class DatasetMeta:
     name: Optional[str]
-    target: str
```

Comment:
should it be `target: Optional[str] = None` ?

Reply:
Hi Andrey, thanks very much for the review!

The `Optional[str]` without default value means that we have to provide it in constructor, but it could be `None`:

```
>>> from giskard.core.core import DatasetMeta
>>> DatasetMeta()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: DatasetMeta.__init__() missing 6 required positional arguments: 'name', 'column_types', 'column_dtypes', 'number_of_rows', 'category_features', and 'target'
```

while having the default value means that we do not have to provide it:

```
>>> from giskard.core.core import DatasetMeta
>>> DatasetMeta()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: DatasetMeta.__init__() missing 5 required positional arguments: 'name', 'column_types', 'column_dtypes', 'number_of_rows', and 'category_features'
```

In the `DatasetMeta` case, all of the information should have been provided by the `Dataset` class, but the target could be `None` now with LLM model.